torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -475,6 +477,18 @@ struct NCCLTraceBuffer {
     return id_++;
   }
 
+  void record_pg(size_t pg_id, std::vector<uint64_t> ranks) {
+    if (!enabled_) {
+      return;
+    }
+    std::lock_guard<std::mutex> guard(mutex_);
+    if (pg_id_to_ranks_.find(pg_id) != pg_id_to_ranks_.end()) {
```

Comment:
When will this happen? or we just need to update the map whenever we call record_pg by pg_id_to_ranks_[pg_id] = ranks;